pyogrio/tests/test_geopandas_io.py

```diff
@@ -186,6 +186,79 @@ def test_read_fids_force_2d(test_fgdb_vsi):
         assert not df.iloc[0].geometry.has_z
 
 
+#@pytest.mark.filterwarnings("ignore: Layer")
+def test_read_sql(naturalearth_lowres):
+    # empty filter should return full set of records
+    df = read_dataframe(naturalearth_lowres, sql=None)
+    assert len(df) == 177
+    assert len(df.columns) == 6
+
+    # should return singular item
+    sql = "SELECT * FROM naturalearth_lowres WHERE iso_a3 = 'CAN'"
+    df = read_dataframe(naturalearth_lowres, sql=sql)
+    assert len(df) == 1
+    assert len(df.columns) == 6
+    assert df.iloc[0].iso_a3 == "CAN"
+    area_canada = df.iloc[0].geometry.area
+
+    # use spatialite function (needs SQLITE dialect)
+    sql = """SELECT ST_Buffer(geometry, 5) AS geometry, name, pop_est, iso_a3 
+               FROM naturalearth_lowres 
+              WHERE ISO_A3 = 'CAN'"""
+    df = read_dataframe(naturalearth_lowres, sql=sql, sql_dialect="SQLITE")
+    assert len(df) == 1
+    assert df.iloc[0].geometry.area > area_canada
+
+    # the geometry column doesn't need/cannot be specified when using the 
+    # default OGRSQL dialect but is returned nonetheless, so 4 columns
+    sql = """SELECT name, pop_est, iso_a3  
```

Comment:
Suggest splitting this into a test that just reads subset of columns and verifies that those are present, and add another test of the `ORDER BY` clause.